pkg/controller/job/job_controller.go

```diff
@@ -802,6 +803,9 @@ func (jm *Controller) manageJob(job *batch.Job, activePods []*v1.Pod, succeeded
 	if rmAtLeast < 0 {
 		rmAtLeast = 0
 	}
+	if rmAtLeast > MaxPodCreateDeletePerSync {
+		rmAtLeast = MaxPodCreateDeletePerSync
+	}
 	podsToDelete := activePodsForRemoval(job, activePods, int(rmAtLeast))
```

Comment:
better crop the array podsToDelete.

Reply:
but modify rmAtLeast before entering activePodForRemoval may reduce calculate operations in that function, why better crop podsToDelete?

Reply:
because activePodsForRemoval might return more than `rmAtLeast`.

Reply:
ah yes, you're right. Thanks for pointing out.